Packet++/src/HttpLayer.cpp

```diff
@@ -855,14 +877,42 @@ HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data,
 		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	std::string codeString = std::string(data + 9, 3);
+	const std::string codeString = std::string(data + 9, 3);
 
 	if(codeString.empty() || (std::find_if(codeString.begin(), codeString.end(), [](unsigned char c){ return !std::isdigit(c); }) != codeString.end()))
 	{
 		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	return HttpResponseStatusCode(std::stoi(codeString));
+	constexpr size_t messageOffset = 13;  // expect "HTTP/x.y XXX YYY", YYY starts from 13
+	size_t offset = messageOffset;
+	bool isMessageFound = false;
+	while(offset < dataLen)
+	{
+		if(data[offset] == '\n')
+		{
+			isMessageFound = true;
+			break;
+		}
+		offset++;
+	}
+
+	if(!isMessageFound)
+	{
+		return HttpResponseStatusCode::HttpStatusCodeUnknown;
+	}
+
+	std::string messageString(data + messageOffset, offset - messageOffset);
+	if(messageString.back() == '\r')
```

Comment:
I found the bug that causes VS build to get stuck. Apparently `back()` cannot be called with an empty string: https://stackoverflow.com/a/35567712

In VS we get this message while running [this test](https://github.com/tigercosmos/PcapPlusPlus/blob/20230830getMessage/Tests/Packet%2B%2BTest/Tests/HttpTests.cpp#L323):

![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/f9a5a3ad-21be-485a-a8e2-c6f62d507c4f)

The solution should be:
```cpp
if(!messageString.empty() && messageString.back() == '\r')
{
	messageString.pop_back();
}

Reply:
@seladb cool, I wonder why linus works fine? Is it a undefined behavior?